rust/src/lib.rs

```diff
@@ -2353,14 +2406,73 @@ pub unsafe extern "C" fn zcashlc_decrypt_and_store_transaction(
     unwrap_exc_or(res, -1)
 }
 
-/// Creates a transaction paying the specified address from the given account.
+fn zip317_helper<DbT>(
+    change_memo: Option<MemoBytes>,
+    use_zip317_fees: bool,
+) -> GreedyInputSelector<DbT, SingleOutputChangeStrategy> {
+    let fee_rule = if use_zip317_fees {
+        StandardFeeRule::Zip317
+    } else {
+        #[allow(deprecated)]
+        StandardFeeRule::PreZip313
+    };
+    GreedyInputSelector::new(
+        SingleOutputChangeStrategy::new(fee_rule, change_memo),
+        DustOutputPolicy::default(),
+    )
+}
+
+/// A struct that contains a pointer to, and length information for, a heap-allocated
+/// boxed slice.
 ///
-/// Returns the row index of the newly-created transaction in the `transactions` table
-/// within the data database. The caller can read the raw transaction bytes from the `raw`
-/// column in order to broadcast the transaction to the network.
+/// # Safety
 ///
-/// Do not call this multiple times in parallel, or you will generate transactions that
-/// double-spend the same notes.
+/// - `ptr` must be non-null and valid for reads for `len` bytes, and it must have an
+///   alignment of `1`. Its contents must be an encoded Proposal protobuf.
+/// - The memory referenced by `ptr` must not be mutated for the lifetime of the struct
+///   (up until [`zcashlc_free_boxed_slice`] is called with it).
+/// - The total size `len` must be no larger than `isize::MAX`. See the safety
+///   documentation of `pointer::offset`.
+#[repr(C)]
+pub struct FfiBoxedSlice {
```

Comment:
If "Its contents must be an encoded Proposal protobuf" shouldn't this have a meaningful name indicating this so that it's not used for other purposes?

Reply:
I think I added that line to the documentation before I decided to implement this as an `FfiBoxedSlice`. I think it's better to just remove that requirement and allow this to contain an arbitrary byte slice, the contents of which is determined by the method it is returned from.